aws-observability/aws-otel-java-instrumentation
# feat: Add Operator compatible image

*Description of changes:*

* Add copy-utility which is a statically linked rust executable that is used to copy the ADOT Java Agent jar into the Operator volume.
* Add docker image compatible with the images used by the OpenTelemetry operator.

To test locally:

```
./gradlew otelagent:build
docker build -t adot-java-operator:latest --build-arg ADOT_JAVA_VERSION=$(./gradlew printVersion -q) .
docker volume create operator-volume
docker run --mount source=operator-volume,dst=/otel-auto-instrumentation adot-java-operator:latest cp /javaagent.jar /otel-auto-instrumentation/javaagent.jar
docker run -ti --mount source=operator-volume,dst=/otel-auto-instrumentation public.ecr.aws/amazonlinux/amazonlinux:latest
Unable to find image 'public.ecr.aws/amazonlinux/amazonlinux:latest' locally
latest: Pulling from amazonlinux/amazonlinux
Digest: sha256:5d5fbc48e357994e2678d338b2dd1fae74d5f4d5ae2c8c0eae60a48a4f95ce38
Status: Downloaded newer image for public.ecr.aws/amazonlinux/amazonlinux:latest
bash-4.2# ls /otel-auto-instrumentation/
javaagent.jar
```

This is equivalent to what is done here https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/instrumentation/javaagent.go#L71


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
